### PR TITLE
Update WW3, Turn on Gaea for waves, Create template for ww3_multi.inp 

### LINKED ISSUE #461: load balance wave tests
## Description
Update load balancing for wave related tests.  

## Solution
It has been requested that tests with waves match the timing of the tests without waves.

### LINKED ISSUE #551: adjust some cpld v16 tests for forecast length and output type
## Description

- The v16 cpld benchmark tests are seen to be coming close to the wall clock time of 30mins when fhmax=12.

- The v16 cpld benchmark test on gaea fails with a pio error (see Issue #509) when the ATM output is the tiled grid. 

## Solution

Reduce the fhmax for the cpld v16 test types to 6 hours.
Change the output type for the cpld v16 tests to ``gaussian_grid``

### PR DESCRIPTION
**Pull request by JessicaMeixner-NOAA**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [ ] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Description

This PR turns on wave tests for S2SW and ATMW on gaea.  It turns on ATMW tests for wcoss-cray as well.  The ww3_multi.inp file was templated.  The S2SW tests were load balanced (#461).  The WW3 input file directory was updated for the creation of the mod_def files to include the building of ww3_grid and a 2 degree mod_def is created as well in case the low resolution test cases want to take advantage of the lower resolution wave model to use even fewer resources for waves. 

In addition we switch the v16 coupled tests to output gaussian because of the pio issue (#509)

* New input directory for WW3 is needed
* New baselines are needed because the number of MPI tasks are changed for S2SW tests and issue 427 has not yet been solved. 

### Issue(s) addressed

- fixes #461
-  provides work-around for issue #509 by setting ``output_grid`` to gaussian. 
- fixes #551 

## Testing

How were these changes tested?  So far on orion, hera and gaea What compilers / HPCs was it tested with? intel Are the changes covered by regression tests? Yes Have regression tests and unit tests (utests) been run?  No On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel (baselines created for all tests but verify against 9 tests failed due to disk quota).
- [x] wcoss_cray
- [x] wcoss_dell_p3

## Dependencies

No dependencies 

co-authors: @DeniseWorthen

### CODE REVIEW

**Comment by MinsukJi-NOAA** [comment]
> @JessicaMeixner-NOAA please bring these up to date with respective authoritative repositories
> 
> * ww3 **NOT** up to date

This notification is incorrect due to a bug (#547) and will be fixed in #527.

File: tests/rt.conf
Comment (DeniseWorthen) [inline tests/rt.conf]: Bin is adding Jet for CPLD and DATM tests in #533. I think that means these ATMW tests could be added to jet also.

**Comment by JessicaMeixner-NOAA** [comment]
New input file for waves is on hera at: /scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/input-data-20210428/WW3_input_data_20210503

File: tests/rt.conf
Comment (JessicaMeixner-NOAA) [reply tests/rt.conf]: I can turn these on, but I have no access to Jet to create the baselines for this, should I go ahead and turn these on?

Comment (DeniseWorthen) [reply tests/rt.conf]: Yes, the new baselines can be created automatically on jet (and cheyenne also) so they can be turned on.

**Comment by DeniseWorthen** [comment]
The directory ``WW3_input_data_20210503`` has been added to ``input-data-20210428`` on all platforms.

**Comment by JessicaMeixner-NOAA** [comment]
The hera.gnu tests passed with existing baselines.  
For hera.intel, I created two new baselines, storing one in save_try1_fv3rt and while the regression tests logs say passed I get the following: 
```
Jessica.Meixner@hfe01:/scratch1/NCEPDEV/stmp4/Jessica.Meixner$ diff -r -q FV3_RT  save_try1_fv3rt
Files FV3_RT/REGRESSION_TEST_INTEL/fv3_regional_quilt_netcdf_parallel/dynf000.nc and save_try1_fv3rt/REGRESSION_TEST_INTEL/fv3_regional_quilt_netcdf_parallel/dynf000.nc differ
Files FV3_RT/REGRESSION_TEST_INTEL/fv3_regional_quilt_netcdf_parallel/dynf024.nc and save_try1_fv3rt/REGRESSION_TEST_INTEL/fv3_regional_quilt_netcdf_parallel/dynf024.nc differ
Files FV3_RT/REGRESSION_TEST_INTEL/fv3_wrtGauss_netcdf_parallel/dynf024.nc and save_try1_fv3rt/REGRESSION_TEST_INTEL/fv3_wrtGauss_netcdf_parallel/dynf024.nc differ
Files FV3_RT/REGRESSION_TEST_INTEL/fv3_wrtGauss_netcdf_parallel/phyf000.nc and save_try1_fv3rt/REGRESSION_TEST_INTEL/fv3_wrtGauss_netcdf_parallel/phyf000.nc differ
Files FV3_RT/REGRESSION_TEST_INTEL/fv3_wrtGauss_netcdf_parallel/phyf024.nc and save_try1_fv3rt/REGRESSION_TEST_INTEL/fv3_wrtGauss_netcdf_parallel/phyf024.nc differ
```

when I do a diff of the two folders.  I'm not sure if this is concerning or not.   I did the same thing on wcoss_cray and there were no differences. 

On gaea, I get failures when running the regression tests in tests that I did not update (the rap, hrrr, etc) tests, but I also got these errors when running from the develop branch last week, so I'm wondering if this is something in my account settings? I can push the logs now, or wait until it's time to run the final set of rt-s as it always ends up being a merge conflict otherwise.

**Comment by DeniseWorthen** [comment]
If I compare the data in those files, they show as being identical. I think the compare_nc tool does a similar compare. Did the ALT compare not give "OK"?

**Comment by JessicaMeixner-NOAA** [comment]
I tried the compare_ncfile.py tool, but the output was a little confusing to me although I think it showed as okay.   This was just to make sure that the changed tests reproduced themselves so the ALT compare is not run to check that in this case.

**Comment by DeniseWorthen** [comment]
I'm not sure we test a new baseline in the same way. What I do is to use rt.sh in create mode and once the baseline is created, I move it to a location. I then edit the ``RTPWD`` variable in rt.sh (LN 418) to point directly to my temporary baseline location. Then I run rt.sh in verify mode which gives me the same report I'd get normally (and it does the compare_nc if it needs to).

**Comment by JessicaMeixner-NOAA** [comment]
@DeniseWorthen thanks! That makes more sense.  I'll try that today.

**Comment by DeniseWorthen** [comment]
We'd like to change the output grid for the cpld v16 tests to use the gaussian grid. This is a temporary workaround for issue #509. We should also increase the wall clock for those tests by 10 minutes to ensure completion (they are running close to the current limit of 30m). I can provide a branch for you to merge or do you want to make the changes directly?

**Comment by JessicaMeixner-NOAA** [comment]
@DeniseWorthen I think merging in a branch is the best idea, do you want to make a PR to my branch or just give me the name of yours to merge in?

**Comment by DeniseWorthen** [comment]
I made a PR to your branch.

**Comment by JessicaMeixner-NOAA** [comment]
@DeniseWorthen @junwang-noaa  I saw that the last commit was just merged.  I'm currently running a test on hera to reduce the size to "only" 79 nodes for the c768 wave coupling test, while this is an improvement we get by taking the settings from the workflow, we likely could reduce this further.  I don't know if you want me to commit that, as it's at least an improvement, leave things as is, or if the next PR is ready, it could go ahead as I continue to work on this, so as to not hold up the commit queue.    Otherwise all testing I have done has passed (gaea, hera.intel, hera.gnu, wcoss-cray) and previous tests on wcoss-dell and orion.   In addition, I have updated the wave tests to run more efficiently in terms of having the wave model take similar time to the atm model, but that did increase the node count.

File: tests/rt.conf
Comment (DeniseWorthen) [inline tests/rt.conf]: I think we want to remove orion from fv3_wrtGauss_nemsio_c768 regardless of any resource improvement you are able to make

Comment (JessicaMeixner-NOAA) [reply tests/rt.conf]: Okay, do I remove it in this commit then?

Comment (DeniseWorthen) [reply tests/rt.conf]: yes, we should remove the 768 orion test.

Comment (JessicaMeixner-NOAA) [reply tests/rt.conf]: Just on orion, or completely?

**Comment by JessicaMeixner-NOAA** [comment]
Good news for the C768 with waves test, it has now been reduced to "only" 40 nodes and ran successfully with 2 threads and 20 TPN on hera.  A big improvement from before, please let me know if further improvements are required/desired.

**Comment by DeniseWorthen** [comment]
That sounds good, we should commit that update for the 768 test which will now only run on hera. 

I also have been able to reduce the cpld v16 tests to 6 hours with the gaussian grid output, which I've tested on Gaea (issue #551 ). The ``cpld_restart_bmarkfrac_v16`` test passes after I changed ``fhzero`` to 3.  This reduces the time required for the cpld v16 non-wave tests to about 15 mins. The settings for the 35d tests are not changed (for consistency w/ workflow). I will make a PR to your branch.

For dell, on this commit we will try to create the baseline but may have to abandon it because of resource issues.

File: tests/default_vars.sh
Comment (junwang-noaa) [inline tests/default_vars.sh:30]: Is the wave on 1 degree grid in the C96 cpl_wave test? It's using half of the total number of tasks.

**Comment by BrianCurtis-NOAA** [comment]
Machine: hera
Compiler: gnu
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/623399322/20210506140041/ufs-weather-model
Please make changes and add the following label back:
hera-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
Machine: hera
Compiler: intel
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/623399322/20210506140043/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_308264
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/623399322/20210506140121/ufs-weather-model
Please make changes and add the following label back:
hera-intel-BL

File: tests/default_vars.sh
Comment (junwang-noaa) [inline tests/default_vars.sh]: This may cause error as FDIAG would be an array of real values while FHZERO is one integer, I'd suggest to use: FHZERO=${FHZERO}

File: tests/default_vars.sh
Comment (JessicaMeixner-NOAA) [reply tests/default_vars.sh:30]: Yes, the wave is the 1 degree in the C96 test. The additional nodes were required for the wave model to run roughly in time with the atm model.

File: tests/default_vars.sh
Comment (DeniseWorthen) [reply tests/default_vars.sh]: Ok, I thought I was being clever. I will change it; I need to see why the hera jobs just stopped anyway.

Comment (DeniseWorthen) [reply tests/default_vars.sh]: Actually @JessicaMeixner-NOAA you need to make this change since it is now your branch we are testing. Thnx.

**Comment by DeniseWorthen** [comment]
> /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/623399322/20210506140041/ufs-weather-model

@BrianCurtis-NOAA This directory doesn't exist. The intel repo directory exists but the rt_308264 doesn't. Can you tell what happened?

File: tests/default_vars.sh
Comment (JessicaMeixner-NOAA) [reply tests/default_vars.sh]: Updated.  Let me know if the fix is incorrect.

**Comment by BrianCurtis-NOAA** [comment]
> > /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/623399322/20210506140041/ufs-weather-model
> 
> @BrianCurtis-NOAA This directory doesn't exist. The intel repo directory exists but the rt_308264 doesn't. Can you tell what happened?

AutoRT deleted it. It shouldn't. After lunch I will dig into it. 
It looks like the baselines never finished.

**Comment by DeniseWorthen** [comment]
The BL job immediately stopped; it doesn't look like it even started?

**Comment by JessicaMeixner-NOAA** [comment]
wcoss-dell baseline has been created, now running against the new baseline

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by BrianCurtis-NOAA** [comment]
Machine: jet
Compiler: intel
Job: BL
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/623399322/20210506174517/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_101822
Test cpld_control_c384 007 failed failed 
Test cpld_control_c384 007 failed in run_test failed 
Please make changes and add the following label back:
jet-intel-BL

**Comment by DeniseWorthen** [comment]
The jet failure was in the baseline creation. I think it was just one of those flakey jet things. I see in err 

```
[225:x471] unexpected DAPL connection event 0x4008 from 284
Fatal error in PMPI_Wait: Internal MPI error!, error stack:
```

@BrianCurtis-NOAA I assume what i need to do is manually move all the baselines that did complete, manually create the missing baseline and then auto-RT. Is that correct?

**Comment by BrianCurtis-NOAA** [comment]
> The jet failure was in the baseline creation. I think it was just one of those flakey jet things. I see in err
> 
> ```
> [225:x471] unexpected DAPL connection event 0x4008 from 284
> Fatal error in PMPI_Wait: Internal MPI error!, error stack:
> ```
> 
> @BrianCurtis-NOAA I assume what i need to do is manually move all the baselines that did complete, manually create the missing baseline and then auto-RT. Is that correct?

Yes. Correct.

**Comment by BrianCurtis-NOAA** [comment]
Machine: jet
Compiler: intel
Job: RT
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/623399322/20210506231511/ufs-weather-model
Please make changes and add the following label back:
jet-intel-RT

**Review by junwang-noaa — APPROVED** [review]
Code changes look good to me, we can commit it when all RTs are done.

**Comment by DeniseWorthen** [comment]
The Jet failure was disk quota. I can see in the rt logs that all jobs that ran passed,  but I can't cat them into a single file. 
But, I don't think all tests ran. I think we're missing 5 tests:

datm_control_cfsr,datm_control_gefs,datm_bulk_gefs,datm_mx025_gefs,datm_cdeps_debug_cfsr 

based on grepping for 'Disk quota exceeded' in the run*.logs

**Comment by DeniseWorthen** [comment]
will merge when ci tests have completed

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by DeniseWorthen — APPROVED** [review]

**Pull request merged**
